src/components/CoursePane/RightPane.js

```diff
@@ -14,10 +16,19 @@ const styles = {
 
 class RightPane extends PureComponent {
     toggleSearch = () => {
-        dispatcher.dispatch({
-            type: 'TOGGLE_SEARCH',
-        });
-        this.forceUpdate();
+        if(RightPaneStore.getFormData().ge != 'ANY' || RightPaneStore.getFormData().deptValue != 'ALL' || 
+            RightPaneStore.getFormData().sectionCode != "" || RightPaneStore.getFormData().instructor != ""){
+            dispatcher.dispatch({
+                type: 'TOGGLE_SEARCH',
+            });
+            this.forceUpdate();
+        }
+        else{
+            openSnackbar(
+                'error',
+                `Please provide one of the following: GE, Department, Course Code or Range, or Instructor`
```

Comment:
Nit: can you flip Department and GE, since department is a more common search.
Also make it `Code/Range`, so we don't have 2 "or"s in there. 
```suggestion
                `Please provide one of the following: Department, GE, Course Code/Range, or Instructor`
```